### PR TITLE
Add env setting to fleet config for on-prem fleets

### PR DESCRIPTION
**Pull request by un-def**
See: https://github.com/dstackai/dstack/issues/1466

### CODE REVIEW

File: src/dstack/_internal/cli/services/configurators/base.py [hunk 1]
```diff
@@ -70,3 +75,29 @@ def register_args(cls, parser: argparse.ArgumentParser):
         args with different parser to show unified help.
         """
         pass
+
+
+class ApplyEnvVarsConfiguratorMixin(BaseApplyConfigurator):
```
Comment (r4victor) [inline src/dstack/_internal/cli/services/configurators/base.py]: Any reason to inherit Mixin from BaseApplyConfigurator? I'd expect Mixin to not inherit from the base class.

Comment (un-def) [reply src/dstack/_internal/cli/services/configurators/base.py]: For type annotations only, no reason from the runtime point of view. I have no strong opinion on this, though.

File: src/dstack/_internal/core/models/common.py [hunk 2]
```diff
@@ -65,7 +65,10 @@ class Config:
     password: Annotated[str, Field(description="The password or access token")]
 
 
-def is_core_model_instance(instance: Any, class_: Type[CoreModel]) -> bool:
+_CM = TypeVar("_CM", bound=CoreModel)
+
+
+def is_core_model_instance(instance: Any, class_: Type[_CM]) -> TypeGuard[_CM]:
```
Comment (r4victor) [inline src/dstack/_internal/core/models/common.py:71]: Didn't know about TypeGuard. That's cool, thanks!

File: src/dstack/_internal/cli/services/configurators/base.py (hunk 1 above)
Comment (r4victor) [reply src/dstack/_internal/cli/services/configurators/base.py]: I see. I'd not depend on super to register mixin args but provide a method like register_env_args() similar to apply_env_vars() and call it from the child explicitly. It's easier to reason about and doesn't require to inherit the mixin.

**Review by r4victor — APPROVED** [review]

File: src/dstack/_internal/cli/services/configurators/base.py (hunk 1 above)
Comment (un-def) [reply src/dstack/_internal/cli/services/configurators/base.py]: Agree, I'll update the PR.

**Review by jvstme — APPROVED** [review]

**Pull request merged**